src/Tooltip.js

```diff
@@ -0,0 +1,130 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import Box from './Box'
+import Card from './Card'
+
+import theme from './theme'
+
+import styled from 'styled-components'
+
+const arrowShadow = props => {
+  return props.top
+    ? {
+        'box-shadow':
+          '-9.66px 9.66px 8px 0 rgba(0,0,0,0.04), -4px 4px 4px 0 rgba(0,0,0,0.08)'
+      }
+    : {
+        'box-shadow':
+          '-1.41px 1.41px 1px 0 rgba(0,0,0,0.01), -3.66px 3.66px 8px 0 rgba(0,0,0,0.04)'
+      }
+}
+
+const arrowAlign = props => {
+  return props.left
+    ? { left: '16px', 'margin-left': props.top ? 0 : '15px' }
+    : props.center
+      ? { left: '50%', 'margin-left': props.top ? '-7px' : '7px' }
+      : { right: '16px', 'margin-right': props.top ? '5px' : '-10px' }
+}
+
+const arrowPosition = props => {
+  return props.top
+    ? {
+        'transform-origin': '0 0',
+        transform: 'rotate(-45deg)',
+        bottom: '-10px'
+      }
+    : {
+        'transform-origin': '0 0',
+        transform: 'rotate(-225deg)',
+        top: '0'
+      }
+}
+
+const arrow = props => {
+  return props.top
+    ? {
+        'transform-origin': '0 0',
+        transform: 'rotate(-45deg)'
+      }
+    : {
+        'transform-origin': '0 0',
+        transform: 'rotate(-225deg)'
+      }
+}
+
+const tooltipPosition = props => {
+  return props.top ? { bottom: '-8px' } : { top: 0 }
+}
+
+const tooltipAlign = props => {
+  return props.right
+    ? { right: 0 }
+    : props.center
+      ? { left: '50%', width: 'auto', transform: 'translateX(-50%)' }
+      : null
+}
+
+const TooltipContent = styled(Box)`
+  display: ${({ isVisible }) => (isVisible ? 'inline' : 'none')};
```

Comment:
Just catching up here – what's the rationale for setting `display: none` instead of not rendering or returning false? If it's for CSS transitions, I think it makes sense to handle in CSS, otherwise it's probably better to not render the HTML at all

Reply:
e.g. something like `{showTooltip && <Tooltip />}` seems like it would keep the API surface area of the design system smaller